analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotCatchSystemException.cs

```diff
@@ -46,7 +47,8 @@ public sealed class DoNotCatchSystemException : SonarDiagnosticAnalyzer
 
                     if (IsSystemException(catchClause.Declaration, c.SemanticModel)
                         && IsCatchClauseEmptyOrNotPattern(catchClause)
-                        && !IsThrowTheLastStatementInTheBlock(catchClause?.Block))
+                        && !IsThrowTheLastStatementInTheBlock(catchClause?.Block)
+                        && c.AzureFunctionMethod() is null)
```

Comment:
I'd move this at the beginning as 1st condition, because this one is the cheapest to evaluate.

Reply:
I moved the checks around a bit from cheapest to most expensive (not measured, just guesswork).